scipy/optimize/tests/test_minpack.py

```diff
@@ -517,6 +517,26 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_covariance(self):
```

Comment:
I saw that you changed `solve_triangular` to `solve_triangular(..., lower=True)` above. It shows that a test with a non-diagonal covariance matrix is sort of necessary.